packages/core/src/components/progress/progress.scss

```diff
@@ -0,0 +1,57 @@
+@use '../../helpers';
+@use '../../mixins';
+
+@mixin Progress() {
+  @if not mixins.includes('Progress') {
+    @include _Progress();
+  }
+}
+
+@mixin _Progress() {
+  .ods-progress {
+    display: inline-grid;
+    grid: auto-flow/auto;
+    width: 100%;
+
+    &.-regular {
+      gap: helpers.space(0.5);
+    }
+
+    &.-large {
+      gap: helpers.space(1);
+    }
+  }
+
+  .ods-progress-background {
+    background-color: helpers.color('background-progress');
+    border-radius: helpers.border-radius('full');
+    overflow: hidden;
+    width: 100%;
+
+    &.-regular {
+      height: helpers.space(0.5);
+    }
+
+    &.-large {
+      height: helpers.space(1);
+    }
+  }
```

Comment:
We can simplify the interface to a single element by using pseudo-elements and CSS vars.

![image](https://user-images.githubusercontent.com/18623773/122389256-aa8f5280-cf68-11eb-8949-a6932ad0761d.png)

`:before` is the background, `:after` is the progressed amount, and by applying text styles directly to `.ods-progress` we don't need the inner `label`.

Makes the vanilla usage much simpler

```html
<div
  class="ods-progress -regular"
  role="progressbar"
  aria-valuenow="25"
  aria-valuemin="0"
  aria-valuemax="100"
  style="--progress: 25%"
>
  25%
</div>
```

Reply:
Way better! Can you and also @josokinas take a look in the new implementation?

Reply:
Looks neat!